source/getting_started/aiot_intelligent_edge/coco_for_applications/end_user_client_applications/android_end_user_client_applications/quick_start_guide_android_end_user_client_apps.rst

```diff
@@ -3,12 +3,332 @@
+            return;
+          }
+   
+          new CocoClient.Configurator()
+              .withCreator(new CreatorEx())
+              .withPlatform(new PlatformInterface() {
+                @Override
+                public String getCwdPath() {
+                  return "provide_current_working_directory";
+                }
+   
+                @Override
+                public String getClientId() {
+                  return "your_client_id_here";
+                }
+   
+                @Override
+                public String getAppAccessList() {
+                  // JSON that describes SDK the capabilities your app will deal
+                  return "{\"appCapabilities\": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}";
+                }
+   
+                @Override
+                public void authCallback(String authorizationEndpoint, String tokenEndpoint) {
+                  // override this callback to perform authentication, This is triggered by SDK when tokens are expired
+                }
+              }).configure();
+        }
+      }
+
+.. important:: 
+
+   ``.withCreator(new CreatorEx())`` is essential as it creates Android
+   flavoured instances of NetworkEx, DeviceEx, ResourceEx, AttributeEx, etc. 
+
+   | Above Ex suffixed classes are sub classes for Network, Device, Resource, Attribute respectively.
+
+----
+
+Authentication
+--------------
+
+1. To login COCO users with their accounts we offer **CocoLoginMgr** a wrapper around **AppAuth-Android** to authenticate users and access their COCO account. The code snippet below achieves Login with COCO using simple result launcher.
+
+   .. code-block:: java
+       :emphasize-lines: 23
+
+        public class MainActivity2 extends AppCompatActivity {
+          private final ActivityResultLauncher<Intent> resultLauncher =
+              registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+                Intent dataIntent = result.getData();
+        
+                if (RESULT_OK != result.getResultCode()) {
+                  // process failure
+                  return;
+                }
+        
+                if (dataIntent.hasExtra(Constants.KEY_AUTH_STATE)) {
+                  String tokens = dataIntent.getStringExtra(Constants.KEY_AUTH_STATE);
+                  // pass tokens to sdk and start making api calls
+                }
+              });
+  
+          @Override
+          protected void onCreate(Bundle savedInstanceState) {
+            Intent loginIntent = new Intent(this, LoginActivity.class)
+                .putExtra(Constants.AUTH_ENDPOINT, authEndpoint)
+                .putExtra(Constants.TOKEN_ENDPOINT, tokenEndpoint);
+        
+            resultLauncher.launch(loginIntent);
+          }
+        }
+
+.. admonition:: Info
+
+    If you are a Third-party-App with its own user base we also have a solution for authentication. :ref:`See <custom_solution_for_self_managed_user_system>` 
```

Comment:
Link doesn't work

Reply:
`:ref:` role is provided by sphinx for cross referencing, so i believe it will not work in github preview.